docs/docs/reference/contextual/derivation.md

```diff
@@ -364,6 +379,35 @@ Assuming the `Ordering.derived` method has a given parameter of type `Mirror[T]`
 compiler generated `Mirror` instance for `Option` and the derivation of the instance will be expanded on the right
 hand side of this definition in the same way as an instance defined in ADT companion objects.
 
+### Automatic Derivation
+
+Automatic derivation of Typeclass instances is less efficient than explicit derivation since it requires a derivation
+to be performed at every site it is invoked. For many use-cases however, this is sufficient. The `derived` appraoch
+above allows for the following implementation.
+
+```scala
+object AutomaticEq {
+  inline implicit def autoEq[T]: Eq[T] = Eq.derived
```

Comment:
This implies that the derivation is performed at each call site and derived code will be duplicated. We should mention that this has potential compile-time/runtime performance implications.

Reply:
Doesn't this need a given `Mirror.Of[T]`?

Reply:
One of the big disadvantages of this approach is that importing this thing smashes all user-defined instances—it seems like that might be worth flagging here.

Reply:
We quickly discuss this point in the weekly meeting, and the conclusion is that we are againts that specific change. We would like do discourage typeclass derivation that expends at every use site, so that pattern shouldn't be exposed in the docs.

Reply:
I agree that maybe it shouldn't be the default way of doing things but isn't it at least worth mentioning that there exists this capability?